src/handlers/register_test.go

```diff
@@ -0,0 +1,137 @@
+package handlers
+
+import (
+	"github.com/Gamify-IT/login-backend/src/gen/db"
+	"github.com/Gamify-IT/login-backend/src/gen/models"
+	"github.com/Gamify-IT/login-backend/src/gen/restapi/operations/register"
+	"github.com/Gamify-IT/login-backend/src/handlers/hash"
+	"golang.org/x/crypto/bcrypt"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestRegisterUser_ShouldReturnOkIfUserNotAlreadyExists(t *testing.T) {
+	// create a new mock
+	// this returns a mock prisma `client` and a `mock` object to set expectations
+	client, mock, ensure := db.NewMock()
+	// defer calling ensure, which makes sure all of the expectations were met and actually called
+	// calling this makes sure that an error is returned if there was no query happening for a given expectation
+	// and makes sure that all of them succeeded
+	defer ensure(t)
+
+	username := "Test User"
+	email := "test@username.com"
+	password := "password"
+
+	hasher := &hash.Mock{FixedReturnValue: []byte("")}
+	passwordHash, _ := hasher.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	stringPasswordHash := string(passwordHash)
+
+	expected := db.UserModel{
+		InnerUser: db.InnerUser{
+			ID:           "user id",
+			CreatedAt:    time.Now(),
+			Name:         username,
+			Email:        email,
+			PasswordHash: stringPasswordHash,
+		},
+	}
+
+	// start the expectation
+	mock.User.Expect(
+		// define your exact query as in your tested function
+		// call it with the exact arguments which you expect the function to be called with
+		// you can copy and paste this from your tested function, and just put specific values into the arguments
+		client.User.FindFirst(
+			db.User.Name.Equals(username),
+		),
+	).Errors(db.ErrNotFound) // sets the object which should be returned in the function call
+
+	mock.User.Expect(
+		client.User.CreateOne(
+			db.User.Name.Set(username),
+			db.User.Email.Set(email),
+			db.User.PasswordHash.Set(stringPasswordHash),
+		),
+	).Returns(expected)
+
+	handler := RegisterUser(client, hasher)
+
+	params := register.NewPostRegisterParams()
+	params.Body = &models.Register{
+		Password: &password,
+		Username: &username,
+		Email:    &email,
+	}
+	params.HTTPRequest = &http.Request{}
+
+	result := handler(params)
+
+	okResult, ok := result.(*register.PostRegisterOK)
+
+	if !ok {
+		t.Errorf("A valid registration should return a 200 status code")
+	}
+	if okResult.Payload.Name != username {
+		t.Errorf("Expected username %q but got %q", username, okResult.Payload.Name)
+	}
+
+	if okResult.Payload.ID != "user id" {
+		t.Errorf("Expected username %q but got %q", username, okResult.Payload.Name)
```

Comment:
```suggestion
		t.Errorf("Expected user ID %q but got %q", userID, okResult.Payload.ID)
```
(assuming `"user id"` gets extracted into `userID`)